google/cloud/bigquery/job.py

```diff
@@ -817,8 +817,9 @@ def result(self, retry=DEFAULT_RETRY, timeout=None):
         """
         if self.state is None:
             self._begin(retry=retry, timeout=timeout)
-        # TODO: modify PollingFuture so it can pass a retry argument to done().
-        return super(_AsyncJob, self).result(timeout=timeout)
+
+        kwargs = {} if retry is DEFAULT_RETRY else {"retry": retry}
```

Comment:
We don't have to follow this pattern here. We just need to bump the minimum `google-api-core` dependency version in `setup.py` and our `constraints-X.txt` file(s).